pkg/packet/bgp/bgp.go

```diff
@@ -11623,6 +11624,98 @@ func NewTunnelEncapSubTLVColor(color uint32) *TunnelEncapSubTLVColor {
 	}
 }
 
+type TunnelEncapSubTLVEgressEndpoint struct {
+	TunnelEncapSubTLV
+	Address net.IP
+}
+
+func (t *TunnelEncapSubTLVEgressEndpoint) DecodeFromBytes(data []byte) error {
+	value, err := t.TunnelEncapSubTLV.DecodeFromBytes(data)
+	if err != nil {
+		return err
+	}
+	if t.Length != 6 && t.Length != 10 && t.Length != 22 {
+		return NewMessageError(BGP_ERROR_UPDATE_MESSAGE_ERROR, BGP_ERROR_SUB_MALFORMED_ATTRIBUTE_LIST, nil, "Not all TunnelEncapSubTLVEgressEndpoint bytes available")
+	}
+	address_family := binary.BigEndian.Uint16(value[4:6])
```

Comment:
What about to use `switch` by address family and in each case make length check? I think it will be more readable.

Reply:
I changed it into a `switch`, and also added named constants to make it more readable.